clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -0,0 +1,650 @@
+        Path to found files and parent directory
+    """
+    for z in Path(path_to_source_data).rglob("*.dcm"):
+        yield str(z), str(Path(z).parent)
+
+
+def filter_dicoms(df: DataFrame) -> DataFrame:
+    """Filters modalities handled by the converter.
+
+    Parameters
+    ----------
+    df_dicom: DataFrame
+        Dataframe containing all of the dicoms available in the source directory.
+
+    Returns
+    -------
+    df_dicom: DataFrame
+        Dataframe with only the modalities handled.
+    """
+    to_filter = [
+        # from GENFI 1
+        "t2_2d_axial",
+        "dwi_trace",
+        "dwi_adc",
+        "dwi_fa",
+        "localizer",
+        "localiser",
+        "nonimage",
+        "dual_pd_t2_axial",
+        "t1_1.25mm_iso",
+        "MoCo",
+        # from GENFI 2
+        "ASL",
+        "motioncorrected",
+        "localiser",
+        "localizer",
+    ]
+
+    df = df.drop_duplicates(subset=["source"])
+    df = df.assign(
+        series_desc=lambda df: df.source_path.apply(
+            lambda x: pdcm.dcmread(x).SeriesDescription
+        )
+    )
+    df = df.assign(
+        acq_date=lambda df: df.source_path.apply(lambda x: pdcm.dcmread(x).StudyDate)
+    )
+    df = df.set_index(["source_path"], verify_integrity=True)
+
+    df = df[~df["source"].str.contains("secondary", case=False)]
+    for file_mod in to_filter:
+        df = df[~df["series_desc"].str.contains(file_mod, case=False)]
+    return df
+
+
+def _check_file(directory: PathLike, pattern: str) -> PathLike:
+    from pathlib import Path
+
+    try:
+        data_file = list(Path(directory).glob(pattern))
+    except StopIteration:
+        raise FileNotFoundError("Clinical data file not found.")
+    if len(data_file) == 0:
+        raise FileNotFoundError("Clinical data not found or incomplete. Aborting")
+    if len(data_file) > 1:
+        raise ValueError("Too many data files found, expected one. Aborting.")
+    return data_file[0]
+
+
+def _read_file(data_file: PathLike) -> pd.DataFrame:
+    import pandas as pd
+
+    return (
+        pd.concat(
+            [
+                pd.read_excel(str(data_file)),
+                pd.read_excel(str(data_file), sheet_name=1),
+            ]
+        )
+        .convert_dtypes()
+        .rename(columns=lambda x: x.lower().replace(" ", "_"))
+    )
+
+
+def find_clinical_data(
+    clinical_data_directory: PathLike,
+) -> List[DataFrame]:
+    """Finds the clinical data associated with the dataset.
+
+    Parameters
+    ----------
+    clinical_data_directory: PathLike
+        Path to the clinical data.
+
+    Returns
+    -------
+    List[DataFrame]
+        Dataframes containing the clinical data
+    """
+    return (
+        _read_file(_check_file(clinical_data_directory, pattern))
+        for pattern in (
+            "FINAL*DEMOGRAPHICS*.xlsx",
+            "FINAL*IMAGING*.xlsx",
+            "FINAL*CLINICAL*.xlsx",
+        )
+    )
+
+
+def complete_clinical_data(
```

Comment:
Since this function performs a merge, I would vote for a more explicit name like `merge_clinical_data`